### ISSUE TITLE
:man_farmer: Performance Rci and Gci build regressions

### ISSUE THREAD

**Issue by Blast545** [ros2/performance_test#34]
Build regressions on the performance jobs `Rci__nightly-performance_ubuntu_focal_amd64` and `Gci__nightly-performance_ubuntu_focal_amd64`. These started happening right after Christmas.

First time happening in each job: 
https://build.ros2.org/view/Gci/job/Gci__nightly-performance_ubuntu_focal_amd64/75
https://build.ros2.org/view/Rci/job/Rci__nightly-performance_ubuntu_focal_amd64/485/

<details><summary>Error details</summary>

```
00:07:10.741 /tmp/ws/install_isolated/fastrtps/include/fastdds/rtps/common/Time_t.h: In function ‘std::istream& eprosima::fastrtps::rtps::operator>>(std::istream&, eprosima::fastrtps::rtps::Time_t&)’:
00:07:10.741 /tmp/ws/install_isolated/fastrtps/include/fastdds/rtps/common/Time_t.h:398:19: warning: conversion to ‘uint32_t’ {aka ‘unsigned int’} from ‘int32_t’ {aka ‘int’} may change the sign of the result [-Wsign-conversion]
00:07:10.741   398 |         t.nanosec(nano);
00:07:10.741       |                   ^~~~
00:07:10.741 Maximum number of instructions (65500) reached
00:07:10.741 Failed to compile '/tmp/ws2/build_isolated/performance_test/src/idlgen/cyclonedds/gen/cyclonedds/Struct32k_.idl'
00:07:10.741 make[2]: *** [src/idlgen/cyclonedds/CMakeFiles/cyclonedds_idl.dir/build.make:178: src/idlgen/cyclonedds/gen/cyclonedds/Struct32k_.c] Error 1
00:07:10.741 make[2]: *** Deleting file 'src/idlgen/cyclonedds/gen/cyclonedds/Struct32k_.c'
00:07:10.741 make[1]: *** [CMakeFiles/Makefile2:653: src/idlgen/cyclonedds/CMakeFiles/cyclonedds_idl.dir/all] Error 2
00:07:10.741 make: *** [Makefile:141: all] Error 2
```
</details>

The error is odd to me, it might be related to the `cyclonedds` idl. Given the time the error started appearing (23-25Dec), I think the build regression came from the new commits added to the `cyclonedds` middleware:
https://github.com/eclipse-cyclonedds/cyclonedds

**Comment by Blast545** [comment]
Probably caused by the check added here: https://github.com/eclipse-cyclonedds/cyclonedds/pull/1077

**Comment by clalancette** [comment]
I've opened an issue to get some advice on how to proceed: https://github.com/eclipse-cyclonedds/cyclonedds/issues/1104

**Comment by clalancette** [comment]
This was fixed by #36, so closing.
